multiversx_sdk/core/transactions_factories/smart_contract_transactions_factory.py

```diff
@@ -88,26 +87,31 @@ def create_transaction_for_execute(self,
                                        arguments: Sequence[Any] = [],
                                        native_transfer_amount: int = 0,
                                        token_transfers: Sequence[ITokenTransfer] = []) -> Transaction:
-        number_of_tokens = len(token_transfers)
+        transfers = list(token_transfers)
+        number_of_tokens = len(transfers)
         receiver = contract
 
         if native_transfer_amount and number_of_tokens:
-            raise BadUsageError("Can't send both native token and custom tokens(ESDT/NFT)")
+            native_tranfer = TokenTransfer.new_from_native_amount(native_transfer_amount)
+            transfers.append(native_tranfer)
+
+            native_transfer_amount = 0
+            number_of_tokens += 1
 
         data_parts: List[str] = []
 
-        if len(token_transfers) == 1:
-            transfer = token_transfers[0]
+        if number_of_tokens == 1:
```

Comment:
If we move `number_of_tokens = len(transfers)` above this line, then maybe we don't need `number_of_tokens += 1` anymore.

Reply:
indeed, but I was using it in the above `if` statement, as well.